book/src/protocol.md

```diff
@@ -0,0 +1,611 @@
+
+#### Witness-extended Emulation
+
+Let $\protocol = \protocol[\group]$ be the interactive argument described above for relation $\relation$ and some group $\group$ with scalar field $\field$. Let $\mathcal{C}$ be a function that takes as input challenges $c_0, c_1, ...$ and univariate polynomials $a'_0(X), a'_1(X), ...$ and produces a satisfying witness for the relation $\relation$ with probability bounded above by $\epsilon \leq \negl$ whenever $g'(\omega^i) = 0$ for all $i \in [0, n)$. We can construct an extractor $\extractor$ such that for any non-uniform algebraic prover $\alg{\prover}$ making at most $q$ queries to its oracle, there exists a non-uniform adversary $\mathcal{F}$ with the property that for any computationally unbounded distinguisher $\mathcal{D}$
+
+$$
+\adv^\srwee_{\ip, \relation}(\alg{\prover}, \distinguisher, \extractor, \sec) \leq q\epsilon + \adv^\dlrel_{\group,n}(\mathcal{F}, \sec).
+$$
+
+_Proof._ We will prove this by invoking Theorem 3 of [[GT20]]((https://eprint.iacr.org/2020/1351)) by describing an extractor function $e$, by defining a set of bad challenges $\badch_{\tr'_c}$ for each possible (valid) partial transcript $\tr'_c$ that contains all messages before the challenge $c$ is sampled, by upper bounding each $|\badch_{\tr'_c}| \leq \epsilon$, and by showing that there exists adversary $\mathcal{F}$ such that
+
+$$
+p_\textnormal{fail}(\protocol, \alg{\prover}, e, \relation) \leq \adv^\dlrel_{\group,n}(\mathcal{F}, \sec)
+$$
+
+where $p_\textnormal{fail}$ represents the probability that $e$ does not return a valid witness.
+
+Recall that a valid transcript is accepting on the condition that
+
+$$
+\sum_{i=0}^{k - 1} [u_j^{-1}] L_j + P' + \sum_{i=0}^{k - 1} [u_j] R_j = [c] \mathbf{G'}_0 + [c \mathbf{b}_0 z] U + [f] W
+$$
+
+and so by taking the representations of each of the prover's commitments with respect to each group element in $\pp$ and rearranging so that they equal zero, we will (efficiently) obtain discrete log relations. Our adversary $\mathcal{F}$ will simulate $\protocol$ to $\alg{\prover}$ and rearrange as stated, and returns the first non-trivial discrete log relation or a trivial relation in the absence of any. This establishes the upper bound on $p_\textnormal{fail}$. In order to complete the proof, we must show the operation of the extractor function $e$ and prove that whenever $\mathcal{F}$ fails to obtain a non-trivial discrete log relation that the extractor function $e$ will obtain a valid witness from the transcript.
```

Comment:
```suggestion
and so by taking the representations of each of the prover's commitments with respect to each group element in $\pp$ and rearranging so that they equal zero, we will (efficiently) obtain discrete log relations. Our adversary $\mathcal{F}$ will simulate $\protocol$ to $\alg{\prover}$ and rearrange as stated, and returns the first non-trivial discrete log relation (or a trivial relation in the absence of any). This establishes the upper bound on $p_\textnormal{fail}$. In order to complete the proof, we must show the operation of the extractor function $e$ and prove that whenever $\mathcal{F}$ fails to obtain a non-trivial discrete log relation that the extractor function $e$ will obtain a valid witness from the transcript.
```